### ISSUE TITLE
关于在其他数据集上跑实验

### ISSUE THREAD

**Issue by wjczf123** [yao8839836/text_gcn#46]
您好，我大概看了下您的程序了，并且在另外一个数据集上用您的方法跑了下这个任务。假设我在一个名为R8的数据集（只是名字叫R8，跟您论文中的R8数据集不同），按照您的组织方式在/data/corpus和data/路径下分别放入R8.txt。data/corpus/R8.txt中数据集采用每行是一个文本，data/R8.txt中采用跟您相同的格式，0   train   one（one是在我这个任务中的类别）等等。想向您请教一个问题：
在我目前看到的情况，因为我把数据集命名为R8，程序需要的两个文档也和您的一致，代码方面好像并不需要改动了。但是在train.py中我打印了test_pred和test_labels（144.145行中定义）。发现test_labels正常，而test_pred在训练过程中，一直没有发生变化，并且对所有样本输出一个类别，或者两个类别。而我的分类任务有12个类别。是因为代码还没有修改好吗。

**Comment by yao8839836** [comment]
@wjczf123 

您好，代码应该不用修改，输出test_pred时训练集和验证集的准确率是否达到较高水平且收敛？

**Comment by wjczf123** [comment]
最后一轮时train_loss=1.71，train_acc= 0.38863 val_loss= 1.74594 val_acc= 0.35915.
Test set results: cost= 1.70858 accuracy= 0.38375。测试集很诡异的全输出了一个类别

**Comment by yao8839836** [comment]
@wjczf123 

训练集的准确率还很低，如果不是学习率太大不能收敛 （可试试调小），我估计就是数据问题。

**Comment by wjczf123** [comment]
哦，行。谢谢您。

**Comment by VolantBoy** [comment]
我也尝试了将您代码迁移到了一份句对分类任务上，验证集性能在0.45569,测试集也很低，您知道为啥嘛

**Comment by yao8839836** [comment]
@VolantBoy 

您好，句对分类是把两个句子拼成一个文档（节点）分类吗？感觉这种任务文本比较短，且需要优先考虑局部单词组合以及词序，而GCN对短文本语料构建的图比较稀疏，且着重强调全局词共现信息，因此表现不好。

**Comment by VolantBoy** [comment]
您好，您所说的GCN对短文本语料构建的图比较稀疏，指的是图上边比较少嘛？您猜测没错，我是将两个句子拼成了一个文档（比较长），然后基于您的代码做的。您的意思是由于没有考虑局部依赖和词序所以性能低。照您的分析来讲，在短文本分类中，局部依赖和词序起到的作用不如全局词的共现嘛？可以解释一下全局词贡献为何起作用的原因嘛？

**Comment by yao8839836** [comment]
@VolantBoy 

您好，是指图上的边比较少。GCN是由于没有考虑局部依赖和词序所以性能低。短文本分类中局部依赖和词序比全局共现要重要，所以CNN， LSTM会表现较好。至于BERT，在预训练的时候已考虑到两个句子关系，所以理应表现好。

**Comment by VolantBoy** [comment]
您好，在短文本分类中局部依赖和词序比全局共现更重要。这个我可以理解，那么在长文本分类中，局部依赖就不重要了吗？全局词共现为何在长文本分类中就有效呢？不是很能理解这个。您可以具体的解释一下嘛？

**Comment by yao8839836** [comment]
@VolantBoy 

您好，长文本分类大多为主题分类，例如文档是说医疗还是军事。一些关键词的组合（共现）比较有区分度，而他们的局部顺序并不是很重要。
